### PR TITLE
Feature-IOPS-2018-Add-Valuesets-Canonical-URL-to-England-HealthcareService

### PR DESCRIPTION
**Pull request by KazeemHamzat**
Feature-IOPS-2018-Add-Valuesets-Canonical-URL-to-England-HealthcareService

### CODE REVIEW

File: codesystems/CodeSystem-England-ERSBookableType.xml
Comment (digitalhealthmartin) [inline codesystems/CodeSystem-England-ERSBookableType.xml:1]: File already added in PR https://github.com/NHSDigital/NHSEngland-FHIR-ImplementationGuide/pull/375/files

File: valuesets/ValueSet-England-DMMedicationRequestControlledDrug.xml
Comment (digitalhealthmartin) [inline valuesets/ValueSet-England-DMMedicationRequestControlledDrug.xml:1]: File already added in PR https://github.com/NHSDigital/NHSEngland-FHIR-ImplementationGuide/pull/375/files

File: valuesets/ValueSet-England-DMPrescriptionTaskStatusReason.xml
Comment (digitalhealthmartin) [inline valuesets/ValueSet-England-DMPrescriptionTaskStatusReason.xml:1]: File already added in PR https://github.com/NHSDigital/NHSEngland-FHIR-ImplementationGuide/pull/375/files

File: valuesets/ValueSet-England-ERSBookableType.xml
Comment (digitalhealthmartin) [inline valuesets/ValueSet-England-ERSBookableType.xml:1]: File already added in PR https://github.com/NHSDigital/NHSEngland-FHIR-ImplementationGuide/pull/375/files

File: valuesets/ValueSet-England-ERSRequestFlowType.xml
Comment (digitalhealthmartin) [inline valuesets/ValueSet-England-ERSRequestFlowType.xml:1]: File already added in PR https://github.com/NHSDigital/NHSEngland-FHIR-ImplementationGuide/pull/375/files

**Review by digitalhealthmartin — APPROVED** [review]

**Review by digitalhealthmartin — CHANGES_REQUESTED** [review]
Raised points about same files added in different PRs

File: codesystems/CodeSystem-England-ERSBookableType.xml
Comment (digitalhealthmartin) [inline codesystems/CodeSystem-England-ERSBookableType.xml:1]: Something looks wrong here as it's gone from ERSPurpose to ERSBookableType (looking at all the changes). Maybe best to delete this file, and create a new file afresh and add the code in it. Hopefully git won't track these sort of changes.

File: codesystems/CodeSystem-England-MedicationRequestEndorsement.xml
Comment (digitalhealthmartin) [inline codesystems/CodeSystem-England-MedicationRequestEndorsement.xml:1]: Add to top line: `<?xml version="1.0" encoding="utf-8"?>`

File: structuredefinitions/England-MedicationRequest.xml [hunk 1]
```diff
@@ -172,6 +172,10 @@
         <profile value="https://fhir.nhs.uk/StructureDefinition/Extension-England-EPSDispensingInformation" />
       </type>
       <isModifier value="false" />
+      <binding>
+        <strength value="required" />
+        <valueSet value="https://fhir.nhs.uk/ValueSet/England-DMPrescriptionEndorsement" />
```
Comment (digitalhealthmartin) [inline structuredefinitions/England-MedicationRequest.xml]: Where does this change come from? Can't find the ticket about it

File: valuesets/ValueSet-England-DMMedicationRequestControlledDrug.xml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        <parameter>
+            <name value="count" />
+            <valueInteger value="2147483647" />
+        </parameter>
+        <parameter>
+            <name value="offset" />
+            <valueInteger value="0" />
+        </parameter>
+        <contains>
+            <system value="https://fhir.nhs.uk/CodeSystem/medicationrequest-controlled-drug" />
```
Comment (digitalhealthmartin) [inline valuesets/ValueSet-England-DMMedicationRequestControlledDrug.xml]: The system value needs updating

File: valuesets/ValueSet-England-DMMedicationRequestControlledDrug.xml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+            <code value="CD3" />
+            <display value="Schedule 3" />
+        </contains>
+        <contains>
+            <system value="https://fhir.nhs.uk/CodeSystem/medicationrequest-controlled-drug" />
```
Comment (digitalhealthmartin) [inline valuesets/ValueSet-England-DMMedicationRequestControlledDrug.xml]: The system value needs updating

File: valuesets/ValueSet-England-DMMedicationRequestControlledDrug.xml [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+            <code value="CD2" />
+            <display value="Schedule 2" />
+        </contains>
+        <contains>
+            <system value="https://fhir.nhs.uk/CodeSystem/medicationrequest-controlled-drug" />
```
Comment (digitalhealthmartin) [inline valuesets/ValueSet-England-DMMedicationRequestControlledDrug.xml]: The system value needs updating

File: valuesets/ValueSet-England-DMMedicationRequestControlledDrug.xml [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+            <code value="CD5" />
+            <display value="Schedule 5" />
+        </contains>
+        <contains>
+            <system value="https://fhir.nhs.uk/CodeSystem/medicationrequest-controlled-drug" />
```
Comment (digitalhealthmartin) [inline valuesets/ValueSet-England-DMMedicationRequestControlledDrug.xml]: The system value needs updating

File: valuesets/ValueSet-England-DMMedicationRequestControlledDrug.xml [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+            <code value="CD4-1" />
+            <display value="Schedule 4 (Part I)" />
+        </contains>
+        <contains>
+            <system value="https://fhir.nhs.uk/CodeSystem/medicationrequest-controlled-drug" />
```
Comment (digitalhealthmartin) [inline valuesets/ValueSet-England-DMMedicationRequestControlledDrug.xml]: The system value needs updating

File: valuesets/ValueSet-England-DMMedicationRequestControlledDrug.xml [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+            <code value="CD4-2" />
+            <display value="Schedule 4 (Part II)" />
+        </contains>
+        <contains>
+            <system value="https://fhir.nhs.uk/CodeSystem/medicationrequest-controlled-drug" />
```
Comment (digitalhealthmartin) [inline valuesets/ValueSet-England-DMMedicationRequestControlledDrug.xml]: The system value needs updating

File: valuesets/ValueSet-England-DMMedicationRequestControlledDrug.xml [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+            <code value="CD1" />
+            <display value="Schedule 1" />
+        </contains>
+        <contains>
+            <system value="https://fhir.nhs.uk/CodeSystem/medicationrequest-controlled-drug" />
```
Comment (digitalhealthmartin) [inline valuesets/ValueSet-England-DMMedicationRequestControlledDrug.xml]: The system value needs updating

File: valuesets/ValueSet-England-DMPrescriptionEndorsement.xml [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+        <parameter>
+            <name value="count" />
+            <valueInteger value="2147483647" />
+        </parameter>
+        <parameter>
+            <name value="offset" />
+            <valueInteger value="0" />
+        </parameter>
+        <contains>
+            <system value="https://fhir.nhs.uk/CodeSystem/medicationrequest-endorsement" />
```
Comment (digitalhealthmartin) [inline valuesets/ValueSet-England-DMPrescriptionEndorsement.xml]: The system value needs updating

File: valuesets/ValueSet-England-DMPrescriptionEndorsement.xml [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+            <code value="CC" />
+            <display value="Contraceptive" />
+        </contains>
+        <contains>
+            <system value="https://fhir.nhs.uk/CodeSystem/medicationrequest-endorsement" />
```
Comment (digitalhealthmartin) [inline valuesets/ValueSet-England-DMPrescriptionEndorsement.xml]: The system value needs updating

File: valuesets/ValueSet-England-DMPrescriptionEndorsement.xml [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+            <code value="FS" />
+            <display value="Sexual Health" />
+        </contains>
+        <contains>
+            <system value="https://fhir.nhs.uk/CodeSystem/medicationrequest-endorsement" />
```
Comment (digitalhealthmartin) [inline valuesets/ValueSet-England-DMPrescriptionEndorsement.xml]: The system value needs updating

File: valuesets/ValueSet-England-DMPrescriptionEndorsement.xml [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+            <code value="ACBS" />
+            <display value="Advisory Committee on Borderline Substances" />
+        </contains>
+        <contains>
+            <system value="https://fhir.nhs.uk/CodeSystem/medicationrequest-endorsement" />
```
Comment (digitalhealthmartin) [inline valuesets/ValueSet-England-DMPrescriptionEndorsement.xml]: The system value needs updating

File: valuesets/ValueSet-England-DMPrescriptionEndorsement.xml [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+            <code value="SLS" />
+            <display value="Selected List Scheme" />
+        </contains>
+        <contains>
+            <system value="https://fhir.nhs.uk/CodeSystem/medicationrequest-endorsement" />
```
Comment (digitalhealthmartin) [inline valuesets/ValueSet-England-DMPrescriptionEndorsement.xml]: The system value needs updating

**Review by digitalhealthmartin — CHANGES_REQUESTED** [review]

File: structuredefinitions/England-MedicationRequest.xml (hunk 1 above)
Comment (KazeemHamzat) [reply structuredefinitions/England-MedicationRequest.xml]: Good spot. Should have mapped ValueSet-England-DMMedicationDispenseType to Extension.extension:dispenseStatus.
Item 4 of the Feedback email.

File: codesystems/CodeSystem-England-ERSBookableType.xml
Comment (KazeemHamzat) [reply codesystems/CodeSystem-England-ERSBookableType.xml:1]: Created a new file, but GitHub sees it the same. 
Have further reviewed with it Declan and considered it okay as long as file content is OK.

File: valuesets/ValueSet-England-DMMedicationRequestControlledDrug.xml
Comment (digitalhealthmartin) [inline valuesets/ValueSet-England-DMMedicationRequestControlledDrug.xml:1]: Remove the added |1.0.0 version from each system value (wasn't there before).

File: valuesets/ValueSet-England-DMPrescriptionEndorsement.xml
Comment (digitalhealthmartin) [inline valuesets/ValueSet-England-DMPrescriptionEndorsement.xml:1]: Remove the added |1.0.0 version from each system value (wasn't there before).

**Review by digitalhealthmartin — CHANGES_REQUESTED** [review]

File: valuesets/ValueSet-England-DMPrescriptionTaskStatusReason.xml [hunk 14]
```diff
@@ -0,0 +1,116 @@
...
+        </include>
+    </compose>
+    <expansion>
+        <identifier value="f61a9b9d-6946-4c29-bf77-06a1a2c21b8c" />
+        <timestamp value="2020-06-01T07:49:44+00:00" />
+        <total value="14" />
+        <offset value="0" />
+        <parameter>
+            <name value="version" />
+            <valueUri value="https://fhir.nhs.uk/CodeSystem/England-MedicationRequestStatusHistory|1.0.0" />
```
Comment (digitalhealthmartin) [inline valuesets/ValueSet-England-DMPrescriptionTaskStatusReason.xml]: Remove |1.0.0

**Review by digitalhealthmartin — CHANGES_REQUESTED** [review]

**Review by digitalhealthmartin — APPROVED** [review]

**Review by SheenaSharma — APPROVED** [review]

**Pull request merged**